### ISSUE TITLE
Question. Why not just use Hashids?

### ISSUE THREAD

**Issue by patrickcurl** [adamhopkinson/laravel-model-hash#6]
I agree this is def, the better way to handle security than ids, and uuid's are way too long. 

but Hashids takes a salt, and you can basically still use the id, but it's hashed to something like GHdfS3 for the url. Curious if this has any benefit over that, other than maybe more choices w/ alphabet?

see: https://github.com/vinkla/laravel-hashids

**Comment by adamhopkinson** [comment]
Good question!

I wasn't aware of hashids when I wrote laravel-model-hash. I do recall the logo, so must have seen it somewhere, but I did a search before writing mine and couldn't find anything that did what I was after. I've now read through the docs, but let me know if I've misunderstood any of it in my answer below.

laravel-model-hash looks to have a different purpose to hashids - primarily that is to replace the id in urls. I could have used hashids to generate those hashes - and this would have removed the need to persist the hash in the database - but I've previously been burned when merging systems with auto-incrementing ids. If the hash is stored in the database, some of this trauma would have been avoided, particularly having to redirect urls when the hash for a record needs to resolve to a new ID (yes I understand that ids should be permanent, but sometimes the world of work isn't ideal!).

In terms of what laravel-model-hash does in addition to what hashids can do:

- it's ready to go out-of-the-box
- it allows full customisation of the hash (alphabet, length, hash name)
- configuration can be taken from the default, from the provided config file or on a model-specific basis
- it automatically switches the RouteKeyName so that the provided hash is used in route model binding instead of the id

None of this is a criticism of hashids - they just have different purposes.
